src/backend/alsa/mod.rs

```diff
@@ -153,17 +153,32 @@ pub struct MidiInputConnection<T: 'static> {
     subscription: Option<PortSubscribe>,
     thread: Option<JoinHandle<(HandlerData<T>, T)>>,
     vport: i32, // TODO: probably port numbers are only u8, therefore could use Option<u8>
-    trigger_send_fd: i32,
+    trigger_send_fd: Option<PipeFd>,
 }
 
 struct HandlerData<T: 'static> {
     ignore_flags: Ignore,
     seq: Seq,
-    trigger_rcv_fd: i32,
+    trigger_rcv_fd: Option<PipeFd>,
     callback: Box<dyn FnMut(u64, &[u8], &mut T) + Send>,
     queue_id: i32, // an input queue is needed to get timestamped events
 }
 
+#[repr(transparent)]
+struct PipeFd(i32);
+
+impl PipeFd {
+    fn get(&self) -> i32 {
+        self.0
```

Comment:
You probably want to assert that `self.0 >= 0` before returning.

Reply:
I don't need to assert that if nobody has changed the value since construction.